Source/LibationSearchEngine/BookRule.cs

```diff
@@ -0,0 +1,21 @@
+﻿using DataLayer;
+using System;
+using System.Collections.ObjectModel;
+
+namespace LibationSearchEngine;
+
+public class BookRule : IIndexRule
+{
+	public FieldType FieldType { get; }
+	public Func<Book, string> ValueGetter { get; }
+	public ReadOnlyCollection<string> FieldNames { get; }
+
+	public BookRule(FieldType fieldType, Func<Book, string> valueGetter, params string[] fieldNames)
```

Comment:
I recommend validation and sanitation on `fieldNames`. Something like

    ArgumentValidator.EnsureNotNull(fieldNames)
    fieldNames = fieldNames.Where(a => !string.IsNullOrWhiteSpace(a)).Select(a => a.Trim())
    if !fieldNames.Any() throw

Reply:
Lemme know if you're intentionally not doing this so I'm not waiting on you before I merge.

Reply:
I'm working on it now.